openhands/agenthub/codeact_agent/tools/search_engine.py

```diff
@@ -0,0 +1,25 @@
+from litellm import ChatCompletionToolParam, ChatCompletionToolParamFunctionChunk
+
+_SEARCH_ENGINE_DESCRIPTION = """Execute a web search query (similar to Google search).
+
+You MUST use this tool as a search engine and find URLs of webpages that are relevant to the search query.
+NOTE: Do NOT use the browser tool to search using Google or Bing since you will be blocked by CAPTCHAs.
```

Comment:
I think we should emphasize that this tool is more suitable for searching, compared to other tools. And we might get rid of the above prompt.

```suggestion
NOTE: When you need to search for information online, please use the `search_engine` tool rather than the `browser` or `web_read` tools. The `search_engine` tool connects directly to a search engine, which will help avoid CAPTCHA challenges that would otherwise block your access.
```

A side-note: `web_read` tool's prompt kinda conflicts with this one: that tool recommends 

> search the webpage content using a Google search query (e.g., url=`https://www.google.com/search?q=YOUR_QUERY`)

which we should remove IFF this tool is enabled. This is, however, out of scope of this PR and needs some design - how shall we deal with dependencies/conflicts between different tools?

Reply:
Updated the description to emphasize that this tool is more suitable for searching and to avoid CAPTCHA challenges. Also noted the conflict with web_read tool - we should handle tool dependencies/conflicts in a future PR.